ludwig/utils/image_utils.py

```diff
@@ -104,62 +66,69 @@ def is_image_score(src_path, img_entry, column: str):
     return 0
 
 
-@functools.lru_cache(maxsize=32)
-def read_image(img: Union[str, bytes, BytesIO, torch.Tensor], num_channels: Optional[int] = None) -> torch.Tensor:
-    """Returns a tensor with CHW format.
+def get_image_read_mode_from_num_channels(num_channels: int) -> ImageReadMode:
+    """Returns the torchvision.io.ImageReadMode corresponding to the number of channels.
 
-    If num_channels is not provided, the image is read in unchanged format. Returns None if the image could not be read.
+    If num_channels is not recognized, returns ImageReadMode.UNCHANGED.
     """
-    if isinstance(img, torch.Tensor):
-        return img
-    if isinstance(img, str):
-        return read_image_from_str(img, num_channels)
-    if isinstance(img, bytes):
-        with BytesIO(img) as buffer:
-            buffer_view = buffer.getbuffer()
-            image_tensor = decode_image(torch.frombuffer(buffer_view, dtype=torch.uint8))
-            del buffer_view
-            return image_tensor
-    if isinstance(img, BytesIO):
-        buffer_view = img.getbuffer()
-        try:
-            image_tensor = decode_image(torch.frombuffer(buffer_view, dtype=torch.uint8))
-            del buffer_view
-            return image_tensor
-        except RuntimeError as e:
-            logger.warning(f"Encountered torchvision error while reading {img}: {e}")
-    logger.warning(f"Could not read image {img}, unsupported type {type(img)}")
+    mode = ImageReadMode.UNCHANGED
+    if num_channels == 1:
+        mode = ImageReadMode.GRAY
+    elif num_channels == 2:
+        mode = ImageReadMode.GRAY_ALPHA
+    elif num_channels == 3:
+        mode = ImageReadMode.RGB
+    elif num_channels == 4:
+        mode = ImageReadMode.RGB_ALPHA
+    return mode
+
+
+def read_image_if_path(item: Any, num_channels: Optional[int] = None) -> Union[Any, torch.Tensor]:
```

Comment:
How important is the "return as-is behavior" here and elsewhere? To me it's more intuitive and self-documenting to return `None` or an exception if the input isn't the type expected, then handle appropriately from the caller.

Reply:
Removed in latest change.